### ISSUE TITLE
False positive UndocumentedPublicClass for functional interfaces

### ISSUE THREAD

**Issue by melihaksoy** [detekt/detekt#3372]
Getting false positive `UndocumentedPublicClass` when converting interface to `fun interface` 

This doesn't generate any issue

```kotlin
/**
 * This interface is an example
 */
interface Example {
    /**
     * Trigger when done
     */
    fun onComplete()
}
```

This ends up failed for `UndocumentedPublicClass`

```kotlin
/**
 * This interface is an example
 */
fun interface Example {
    /**
     * Trigger when done
     */
    fun onComplete()
}
```

**Comment by cortinico** [comment]
I've added a test here #3374 and seems like your snippet is working fine. 

Are you on the latest version of Detekt/Kotlin? 
Can you provide the full detekt report and a reproducer project?

**Comment by melihaksoy** [comment]
Can confirm that it works on my side project as well.

Just tried again in the project I'm having problems with - it initially had `1.10.0`, updated to `1.15.0`. Still getting the error there 🤔 Weird.

I'm not really sure what may be causing this, change I made is exactly as above - just converting it to `fun interface` and getting `UndocumentedPublicClass` 🤷

**Comment by cortinico** [comment]
> Can confirm that it works on my side project as well.

"it works", you mean it's bugged right.

>  just converting it to fun interface and getting UndocumentedPublicClass 🤷

Are you on the latest version of Kotlin?
Can you create a reproducer project?

**Comment by melihaksoy** [comment]
Sorry, I meant it works as intended and doesn't generate the error on showcase I wanted to reporduce issue.

Weird enough, it is still a false positive on the production project I'm working on.

They got same kotlin ( `1.4.21`) & detekt (`1.15.0`) versions and interface is exactly same as above. Is there anything I can check that might be causing it ?

**Comment by cortinico** [comment]
> Is there anything I can check that might be causing it ?

Sadly, unless you link your project or a reproducer project, there is not much we can do to help.

**Comment by schalkms** [comment]
Closed due to 
https://github.com/detekt/detekt/issues/3372#issuecomment-760264211
